### PR TITLE
add bookDetail

### PR DESCRIPTION
**Pull request by ebedoyaalzate**
## Summary

- Add book detail page

## Screenshots

Screen Shot 2022-03-17 at 11 20 16 AM

## Trello Card

https://trello.com/c/Rx9Ucirg/59-book-detail

### CODE REVIEW

File: src/components/BookCard/index.tsx [hunk 1]
```diff
@@ -10,11 +13,11 @@ type BookCardProp = {
 
 function BookCard({ book }: BookCardProp) {
   return (
-    <div className={styles.book}>
+    <Link to={`${routes.auth.home}/${book.id}`} className={styles.book}>
```
Comment (ivansevillaa) [inline src/components/BookCard/index.tsx:16]: Maybe is a good idea to have a constants file with the paths we have in the app. And then use that constant here.

File: src/services/books.ts [hunk 2]
```diff
@@ -1,4 +1,12 @@
 import api from '../config/api';
-import { BookListResponse } from '../types/book';
+import { BookListResponse, Book } from '../types/book';
 
 export const getBooks = () => api.get<BookListResponse>('/books');
+
+export const getBookDetail = async (id: string): Promise<Book> => {
```
Comment (leonardofrp5) [inline src/services/books.ts]: react-query resolves promises, using this like this is like you're not properly using react-query right

 query resexport const getBook = (id: string) => headers.get<Book>(`/books/${id}`, {});
```suggestion
export const getBookDetail = (id: string) => api.get<Book>(`/books/${id}`, {});
```

File: src/screens/BookDetail/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    height: 300px;
+    margin: 20px;
+    object-fit: contain;
+    width: 200px;
+  }
+}
+.book-info {
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
```
Comment (leonardofrp5) [inline src/screens/BookDetail/styles.module.scss]: sort

**Review by leonardofrp5 — CHANGES_REQUESTED** [review]

File: src/services/books.ts (hunk 2 above)
Comment (ebedoyaalzate) [reply src/services/books.ts]: I'm trying to use react query like the first getBooks but always get an error with the types in the useQuery call, this is an example that I try but I always get an error
Screen Shot 2022-03-17 at 10 02 51 AM
Screen Shot 2022-03-17 at 10 03 43 AM
.

Comment (leonardofrp5) [reply src/services/books.ts]: `if you can try const { isLoading, data, isError } = useQuery(['bookDetail', id], () => getBookDetail(id));`

in this file src/screens/BookDetail/index.tsx

**Review by leonardofrp5 — APPROVED** [review]

**Pull request merged**